core/src/main/java/org/archifacts/core/model/ArtifactContainer.java

```diff
@@ -67,18 +67,20 @@ public boolean equals(final Object obj) {
 
 	@Override
 	public Set<ArtifactRelationship> getOutgoingRelationships() {
-		return getArtifacts()
-				.stream()
-				.flatMap(a -> a.getOutgoingRelationships().stream())
-				.collect(Collectors.toCollection(LinkedHashSet::new));
+		return Collections.unmodifiableSet(
```

Comment:
Two alternative approaches to avoid the type cast:

- Use a temporary (final) variable to store the set. 
- Use `.collect(collectingAndThen(toCollection(LinkedHashSet::new), Collections::unmodifiableSet));` instead.

However, I don't care that much about the type cast, so the current solution is fine for me as well. But please change the other code as well if you decide to change this.

Reply:
Thanks for the hint. I did not like that code as well and intrudced `org.archifacts.core.model.ArchifactsCollectors.toUnmodifiableLinkedSet()` as we need this behaviour a lot in the model.